noxfile.py

```diff
@@ -147,6 +147,30 @@ def run_pytest(
     )
 
 
+def get_test_files(root: str, ignore_paths: Sequence[str]) -> list[str]:
```

Comment:
i don't like this, it is very prone to break if we change something about our tests naming, also the convention by default is to support both `test_*` and `*_test` see here: https://docs.pytest.org/en/stable/explanation/goodpractices.html#conventions-for-python-test-discovery

could you please change it to use pytest to actually collect all the relevant files?

Reply:
better yet, i don't even think we need all these environments, seems like you can do all of it directly in the config file!